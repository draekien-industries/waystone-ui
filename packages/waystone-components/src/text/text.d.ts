/** @jsxImportSource theme-ui */
import React, { FC } from 'react';
import { BoxProps } from 'theme-ui';
import { TextVariant } from '@waystone/core';
import { HasColor } from '../common/interfaces';
/**
 * Props for rendering the {@link Text} component.
 * @extends React.HTMLAttributes<HTMLDivElement>
 * @extends BoxProps
 */
export interface TextProps extends Omit<React.HTMLAttributes<HTMLDivElement>, 'color'>, Omit<BoxProps, 'color'>, HasColor {
    /**
     * Should the text component place itself on the same line
     * as the preceeding component (if possible).
     * @default false
     */
    inline?: boolean;
    /**
     * The text variant to render.
     * @override
     * @default 'body'
     */
    variant?: TextVariant;
}
/**
 * Renders a text component.
 * @returns The text component
 */
export declare const Text: FC<TextProps>;
//# sourceMappingURL=text.d.ts.map