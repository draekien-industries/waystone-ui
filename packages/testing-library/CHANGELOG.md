# @waystone/testing-library

## 9.2.0

### Minor Changes

- 21db598: chore: bump deps and add documentation

### Patch Changes

- Updated dependencies [21db598]
  - @waystone/theme@9.3.0

## 9.1.6

### Patch Changes

- 41664e2: fix: use type imports to improve bundling
- Updated dependencies [41664e2]
  - @waystone/theme@9.2.5

## 9.1.5

### Patch Changes

- 856ba2b: fix: missing dependencies when installing package
- Updated dependencies [856ba2b]
  - @waystone/theme@9.2.4

## 9.1.4

### Patch Changes

- 5b726aa: fix build and dependencies
- Updated dependencies [5b726aa]
  - @waystone/theme@9.2.3

## 9.1.3

### Patch Changes

- 7086019: Update dependencies
- Updated dependencies [7086019]
  - @waystone/theme@9.2.2

## 9.1.2

### Patch Changes

- @waystone/theme@9.2.1

## 9.1.1

### Patch Changes

- Updated dependencies [8f75dd6]
  - @waystone/theme@9.2.0

## 9.1.0

### Minor Changes

- 5c43357: Switch to pnpm and implement minimal syntax highlighter

### Patch Changes

- Updated dependencies [5c43357]
  - @waystone/theme@9.1.0

## 9.0.0

### Major Changes

- 4a19c42: Update package structure

  Created the `@waystone/ui` package for easier installation and created the `@waystone/testing-library`
  package for a preconfigured testing-library render function.

  Uninstall your `@waystone/*` packages, and run the command `yarn add @waystone/ui`
  to update to the latest version. You will need to update the imports to point
  to the new source.

### Patch Changes

- Updated dependencies [4a19c42]
  - @waystone/theme@9.0.0
