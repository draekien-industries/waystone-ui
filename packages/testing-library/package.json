{
  "name": "@waystone/testing-library",
  "version": "10.0.2",
  "private": true,
  "main": "./src/index.ts",
  "repository": {
    "type": "git",
    "url": "https://github.com/draekien-industries/waystone-ui"
  },
  "description": "The react component library used by draekien-industries",
  "author": "draekien-industries",
  "sideEffects": false,
  "license": "MIT",
  "files": [
    "src/**"
  ],
  "scripts": {
    "lint": "eslint --ext .tsx,.ts,.jsx,.js ./src/ --fix",
    "lint:release": "eslint --ext .tsx,.ts,.jsx,.js ./src/"
  },
  "devDependencies": {
    "@waystone/tsconfig": "workspace:*",
    "eslint-config-waystone": "workspace:*",
    "eslint-plugin-waystone": "workspace:*",
    "react": "^18.3.1",
    "react-dom": "^18.3.1"
  },
  "dependencies": {
    "@testing-library/jest-dom": "^6.6.3",
    "@testing-library/react": "^16.2.0",
    "@testing-library/user-event": "^14.6.0",
    "@theme-ui/color": "^0.17.1",
    "@theme-ui/css": "^0.17.1",
    "@waystone/theme-provider": "workspace:*",
    "jest": "^29.7.0",
    "jest-environment-jsdom": "^29.7.0",
    "theme-ui": "^0.17.1"
  },
  "publishConfig": {
    "access": "public"
  }
}
