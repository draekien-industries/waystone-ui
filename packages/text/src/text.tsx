import type { ColorAttributes, VariantAttributes } from '@waystone/types';
import type { ThemeFontVariants } from '@waystone/theme';
import type { ComponentProps, ElementType } from 'react';
import { Text as ThemeUIText } from 'theme-ui';

export const textVariants = [
  'hero',
  'heading',
  'title',
  'subtitle',
  'eyebrow',
  'body',
  'caption',
  'small',
  'monospace',
] as const satisfies Array<ThemeFontVariants>;

export type TextProps<TAs extends ElementType> = {
  as?: TAs;
  inline?: boolean;
} & VariantAttributes<ThemeFontVariants> &
  ColorAttributes &
  Omit<ComponentProps<TAs>, 'color'>;

export const Text = <TAs extends ElementType>({
  children,
  inline,
  variant,
  color,
  ...rest
}: TextProps<TAs>) => (
  <ThemeUIText
    sx={{
      display: inline ? 'inline-block' : 'block',
      variant: `text.${variant}`,
      color: color || 'inherit',
      margin: inline ? 0 : undefined,
    }}
    {...rest}>
    {children}
  </ThemeUIText>
);
