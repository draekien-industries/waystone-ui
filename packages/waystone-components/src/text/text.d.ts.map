{"version":3,"file":"text.d.ts","sourceRoot":"","sources":["text.tsx"],"names":[],"mappings":"AAAA,gCAAgC;AAChC,OAAO,KAAK,EAAE,EAAE,EAAE,EAAE,MAAM,OAAO,CAAC;AAClC,OAAO,EAAuB,QAAQ,EAAE,MAAM,UAAU,CAAC;AACzD,OAAO,EAAE,WAAW,EAAE,MAAM,gBAAgB,CAAC;AAE7C,OAAO,EAAE,QAAQ,EAAE,MAAM,sBAAsB,CAAC;AAEhD;;;;GAIG;AACH,MAAM,WAAW,SACf,SAAQ,IAAI,CAAC,KAAK,CAAC,cAAc,CAAC,cAAc,CAAC,EAAE,OAAO,CAAC,EACzD,IAAI,CAAC,QAAQ,EAAE,OAAO,CAAC,EACvB,QAAQ;IACV;;;;OAIG;IACH,MAAM,CAAC,EAAE,OAAO,CAAC;IACjB;;;;OAIG;IACH,OAAO,CAAC,EAAE,WAAW,CAAC;CACvB;AAED;;;GAGG;AACH,eAAO,MAAM,IAAI,EAAE,EAAE,CAAC,SAAS,CAU9B,CAAC"}