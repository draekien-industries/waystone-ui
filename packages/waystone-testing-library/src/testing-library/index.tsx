'use client';

import type { RenderOptions, RenderResult } from '@testing-library/react';
import { render } from '@testing-library/react';
import { Waystone } from '@waystone/theme-provider';
import type { PropsWithChildren } from 'react';
import type React from 'react';
import { useEffect } from 'react';
import { useColorMode } from 'theme-ui';

const TestProvider = ({ children }: PropsWithChildren) => (
  <Waystone>{children}</Waystone>
);

const DarkModeWrapper = ({ children }: PropsWithChildren) => {
  const [, setColorMode] = useColorMode();

  useEffect(() => {
    setColorMode('dark');
  }, [setColorMode]);

  return <div>{children}</div>;
};

const themedRender: (
  ui: React.ReactElement,
  options?: RenderOptions
) => RenderResult = (ui, options?) =>
  render(ui, { wrapper: options?.wrapper || TestProvider, ...options });

export * from '@testing-library/react';
export * from '@testing-library/user-event';

export { DarkModeWrapper, themedRender as render };
