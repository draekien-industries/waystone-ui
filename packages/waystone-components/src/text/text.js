var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import { jsx as _jsx } from "theme-ui/jsx-runtime";
import { Text as ThemeUiText } from 'theme-ui';
import { textCss } from './text.styles';
/**
 * Renders a text component.
 * @returns The text component
 */
export const Text = (_a) => {
    var { children, inline = false, variant = 'body', color } = _a, rest = __rest(_a, ["children", "inline", "variant", "color"]);
    return (_jsx(ThemeUiText, Object.assign({ sx: textCss({ inline, variant, color }) }, rest, { children: children })));
};
