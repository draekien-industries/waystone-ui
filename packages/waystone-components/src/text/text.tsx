/** @jsxImportSource theme-ui */
import React from 'react';
import { Text as ThemeUiText } from 'theme-ui';
import { TextVariant } from '@waystone/core';
import { textCss } from './text.styles';
import { HasColor, HasVariant } from '../common/interfaces';

/**
 * Props for rendering the {@link Text} component.
 * @extends HasColor
 * @extends HasVariant
 */
export interface TextProps extends HasColor, HasVariant {
  /**
   * The type of element to render onto the DOM.
   * @default 'span'
   */
  as?: React.ElementType;
  /**
   * Should the text component place itself on the same line
   * as the preceeding component (if possible).
   * @default false
   */
  inline?: boolean;
  /**
   * The text variant to render.
   * @override
   * @default 'body'
   */
  variant?: TextVariant;
  /**
   * The components to render as children.
   */
  children: React.ReactNode;
}

/**
 * A component for rendering text onto the DOM with props for setting the text variant, color,
 * HTML element type.
 */
export const Text = ({
  children,
  as = 'span',
  inline = false,
  variant = 'body',
  color,
  ...rest
}: TextProps) => (
  <ThemeUiText sx={textCss({ inline, variant, color })} as={as} {...rest}>
    {children}
  </ThemeUiText>
);
