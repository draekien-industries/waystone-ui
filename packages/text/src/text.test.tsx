import { render } from '@waystone/testing-library';
import { textVariants, Text } from './text';

describe('<Text />', () => {
  describe.each(textVariants)('variant %s', (variant) => {
    it('should match snapshot', () => {
      const { container } = render(<Text variant={variant}>Lorem Ipsum</Text>);

      expect(container).toMatchSnapshot();
    });
  });
});
