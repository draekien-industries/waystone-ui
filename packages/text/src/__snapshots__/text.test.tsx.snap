// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> variant body should match snapshot 1`] = `
<div>
  <span
    class="css-1k3tifs-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant caption should match snapshot 1`] = `
<div>
  <span
    class="css-1u0r8zu-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant eyebrow should match snapshot 1`] = `
<div>
  <span
    class="css-13k5pct-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant heading should match snapshot 1`] = `
<div>
  <span
    class="css-epqv3n-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant hero should match snapshot 1`] = `
<div>
  <span
    class="css-1mcj2lz-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant label should match snapshot 1`] = `
<div>
  <span
    class="css-ejy0wb-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant monospace should match snapshot 1`] = `
<div>
  <span
    class="css-1s1v8lo-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant small should match snapshot 1`] = `
<div>
  <span
    class="css-4bk1g8-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant subtitle should match snapshot 1`] = `
<div>
  <span
    class="css-5rq687-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;

exports[`<Text /> variant title should match snapshot 1`] = `
<div>
  <span
    class="css-1bsil5k-Text"
  >
    Lorem Ipsum
  </span>
</div>
`;
